components/snap_recovery/src/services.rs

```diff
@@ -233,7 +236,6 @@ impl<ER: RaftEngine> RecoverData for RecoveryService<ER> {
                 }
             }
 
-            let mut rxs = Vec::with_capacity(leaders.len());
             for &region_id in &leaders {
                 if let Err(e) = raft_router.send_casual_msg(region_id, CasualMessage::Campaign) {
```

Comment:
How about moving the logic into leader_keeper?

Reply:
Sounds good, let me try it.